### DISCUSSION TITLE
Self-hosting: which config is used if a repo does not have renovate.json file

### DISCUSSION THREAD

**Discussion by amimas** [renovatebot/renovate#10173] [Request Help] [answered]
I'm setting up a self-hosted bot. I have the following configurations:

```js
require('dotenv').config()

module.exports = {
  /* Self Hosted configurations */

  platform: 'gitlab',  /* https://docs.renovatebot.com/self-hosted-configuration/#platform */
  endpoint: '****/api/v4/',
  token: process.env.GITLAB_USER_TOKEN,

  autodiscover: true,  /* https://docs.renovatebot.com/self-hosted-configuration/#autodiscover */
  onboarding: false,  /* https://docs.renovatebot.com/self-hosted-configuration/#onboarding */
  requireConfig: false,  /* https://docs.renovatebot.com/self-hosted-configuration/#requireconfig */
  
}
```

The above config will avoid onboarding process and scan discovered repositories even if it doesn't have a confiig file (i.e. `renovate.json`).

I believe in a repo's `renovate.json` we can set various settings and/or extend config presets. But, if a repository does not have a config file, what settings will be used when it is processed? Is the following default preset used in this case?

https://docs.renovatebot.com/presets-config/#configbase

**Comment by rarkins** [comment, marked as answer]
It will use the default values you see in https://docs.renovatebot.com/configuration-options/ unless configured otherwise in your `module.exports`. The `config:base` preset is not the default - it needs to be opted into using the `extends` array.

**Reply by amimas** [reply]
Gotcha. That makes sense. So, in my case, I can  add `extends: [ 'config:base' ]` into my `config.js` and that default preset will used, right?

**Reply by amimas** [reply]
Sorry for re-opening this question but it seems what I mentoined above will not work according to the docs.

https://docs.renovatebot.com/configuration-options/#extends

> Configuration presets to use/extend. Note: does not work if configured in config.js.

So in my scenario, I cannot set my own "defaults" for all repositories. The repositories will have to define use the `extends` or other configurations in its own config file (i.e. `renovate.json`). Please let me know if I missed something.

**Reply by rarkins** [reply]
Actually we made improvements so that it does work in `config.js` now, although I recommend you push as much config as you an into repos, not the bot config.

**Reply by amimas** [reply]
Oh okay. Good to know. Thanks for the confirmation that `extends` can now be used in `config.js`.

Yes, I agree that these configs should be in the repositories (i.e. `renovate.json`). I'm just looking to provide a "default" since I will not be using the onboarding process. This is to ensure repos without a `renovate.json` file has a default config.
